internal/cgrouputils/register.go

```diff
@@ -0,0 +1,39 @@
+package cgrouputils
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/sys/unix"
+)
+
+// Memory cgroup root path
+const (
+	MemRoot string = "/sys/fs/cgroup/memory"
+)
+
+// RegisterMemoryThreshold registers an eventfd(2) and a memory threshold in
+// 'cgroup.event_control' of a cgroup. The eventfd will trigger when
+// memory.usage_in_bytes exceeds the threshold set.
+// https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt Section 9
+func RegisterMemoryThreshold(cgPath string, threshold int64) (*os.File, error) {
```

Comment:
Recommend breaking the code up a bit into stanzas: One stanza for setting up `efd`, one for `f`, and then one that does the writing.

Reply:
Are you referring to different functions or just separation of logic indicated by newlines as below 
`go
efd, err := unix.Eventfd(0, unix.EFD_CLOEXEC)
	if err != nil {
		return nil, err
	}
	efdFile := os.NewFile(uintptr(efd), "efd")
	defer func() {
		if err != nil {
			efdFile.Close()
		}
	}()
        //newline added
	f, err := os.Open(filepath.Join(cgPath, "memory.usage_in_bytes"))
	if err != nil {
		return nil, err
	}
	defer f.Close()
	// newline added
	data := fmt.Sprintf("%d %d %d", efdFile.Fd(), f.Fd(), threshold)
	ecPath := filepath.Join(cgPath, "cgroup.event_control")
	if err := ioutil.WriteFile(ecPath, []byte(data), 0700); err != nil {
		efdFile.Close()
		return nil, err
	}
	return efdFile, err
`

Reply:
don't know why that one line is non indented lol

Reply:
Just separate them with blank lines please.

Reply:
Ok cool. The comments were added just to make it more obvious, they weren't going to be kept if that wasn't clear